server/rooms.ts

```diff
@@ -179,6 +180,7 @@ export abstract class BasicRoom {
 	scavLeaderboard: AnyObject;
 	giveaway?: QuestionGiveaway | LotteryGiveaway | null;
 	gtsga?: GTSGiveaway | null;
+	mastermindGame?: Mastermind | null;
```

Comment:
Is there any reason why this needs to be directly in room, instead of just `room.game.mastermindGame` or something?

Reply:
Well, `room.game` represents a round of Mastermind here, so `mastermindGame` needs to exist while `room.game`  is null.

Reply:
Would it be very hard to do it the other way around? Where `room.game` is the entire Mastermind game, and `room.game.round` is a round of Mastermind?

Reply:
That would also fix your `RoomGamePlayer` issue.

Reply:
That would be _possible_ but probably overcomplicated, since Mastermind rounds are just slightly modified Trivia games, and we still want Trivia to extend RoomGame.

Reply:
I... would still massively prefer that to the weirdness going on in this PR which the RoomGame system was never designed to support.

If this is a frequent problem, I think the correct way to do this would be to spec out a `.subGame` property on roomgames which can take `RoomGame` instances.

Reply:
Either way, a simple solution is just not using RoomGame the way it was designed to be used and would need to be carefully audited and end up overcomplicated no matter what.

Reply:
This uses `Room#subGame` now.